js/battle.js

```diff
@@ -6654,10 +6654,16 @@ var Battle = (function () {
 			poke.healthParse(args[3]);
 			poke.removeVolatile('itemremoved');
 			if (args[0] === 'switch') {
-				if (poke.side.active[slot]) {
-					poke.side.switchOut(poke.side.active[slot]);
+				if (poke.side.battle.speciesClause &&
+						poke.side.active[slot] &&
+						poke.side.active[slot].ident === poke.ident) {
+					poke.side.replace(poke);
+				} else {
+					if (poke.side.active[slot]) {
+						poke.side.switchOut(poke.side.active[slot]);
+					}
+					poke.side.switchIn(poke);
```

Comment:
Good idea, but this probably doesn't need this complicated if statement here. You should probably just test Species Clause directly in `getPokemon`.